chain-signatures/contract/src/lib.rs

```diff
@@ -331,6 +331,34 @@ impl VersionedMpcContract {
         }
     }
 
+    #[handle_result]
+    pub fn revoke_join(&mut self) -> Result<(), Error> {
+        log!("revoke_join: signer={}", env::signer_account_id());
+        let protocol_state = self.mutable_state();
+
+        match protocol_state {
+            ProtocolContractState::Running(RunningContractState {
+                candidates,
+                join_votes,
+                ..
+            }) => {
+                let signer_account_id = env::signer_account_id();
+                if candidates.get(&signer_account_id).is_none() {
+                    return Err(JoinError::RevokeNotCandidate.into());
+                }
+
+                // cleanup the existing votes
+                join_votes.remove(&signer_account_id);
+
+                // remove from candidates
+                candidates.remove(&signer_account_id);
```

Comment:
we should not be removing the candidate at all here. This should purely revoke the join vote. Candidates themselves can update their own info through `join`

Reply:
actually, you're trying to remove the candidacy. It's better to name this to be `remove_candidacy`. Revoke join feels like it would be removing just the votes